site/en/blog/fedcm-auto-reauthn-origin-trial/index.md

```diff
@@ -0,0 +1,177 @@
+already previously acknowledged. Therefore we are introducing a more streamlined
+UX that RPs can choose for their returning users.
+
+[FedCM auto-reauthentication](https://github.com/fedidcg/FedCM/issues/429)
+("auto-reauthn" in short) reauthenticates users automatically (when RPs opt-in),
+when they come back after their initial authentication using FedCM. "The initial
+authentication" here means the user creates an account or signs into the RP's
+website by tapping on the **"Continue as..."** button on FedCM's sign-in dialog
+for the first time on the same browser instance.
+
+<figure style="width: 300px; margin: auto; margin-top: 2em;">
+  {% Img
+    src="image/YLflGBAPWecgtKJLqCJHSzHqe2J2/jQvYTB6sHw5jGshyRxn6.png",
+    alt="A dialog the user taps on to create an account or to authenticate.", width="800", height="566", class="screenshot"
+  %}
+  <figcaption>A dialog the user taps on to create an account or to authenticate.</figcaption>
+</figure>
+
+The RP can request auto-reauthn by calling `navigator.credentials.get()` with `autoReauthn: true`.
+
+```js
+const cred = await navigator.credentials.get({
+  identity: {
+    providers: [{
+      configURL: "https://idp.example/fedcm.json",
+      clientId: "1234",
+    }],
+    // NOTE: We are exploring different API options to expose this
+    // functionality here:
+    // https://github.com/fedidcg/FedCM/issues/429#issuecomment-1426358523
+    // You should expect that, as a result of the origin trial, we'll
+    // learn more from developers and browser vendors what works best here.
+    autoReauthn: true, // default to false
+  },
+});
+
+```
+
+With this call, auto-reauthn happens under the following conditions:
+
+* FedCM is available to use. For example, the user has not disabled FedCM either
+  globally or for the RP.
+* The user used only one FedCM account to sign into the website on this browser.
+* The user is signed into the IdP with that account.
+* The auto-reauthn didn't happen within the last 10 minutes.
+
+When the above conditions meet, an attempt to automatically reauthenticate the
+user starts as soon as the FedCM `navigator.credentials.get()` is invoked.
+
+
+<figure style="width: 300px; margin: auto; margin-top: 2em;">
+  {% Video
+    src="video/YLflGBAPWecgtKJLqCJHSzHqe2J2/eYnQxoDMNh4rpsyjcsQI.mp4",
+    autoplay="true", loop="true", class="screenshot"
+  %}
+  <figcaption>A user is auto-reauthenticating to an RP using FedCM</figcaption>
+</figure>
+
+It would be annoying if the user is auto reauthenticated immediately after the
+user signs out.  Developers are recommended to design the sign-out flow not to
+make this happen.
+
+{% Aside %}
+
+As alluded earlier, FedCM has a 10-minute quiet period after an auto-reauthn to
+prevent this behavior. We are also exploring other approaches to achieve this
+such as using
+[`CredentialsContainer.preventSilentAccess()`](https://developer.mozilla.org/docs/Web/API/CredentialsContainer/preventSilentAccess).
+
+{% endAside %}
```

Comment:
```suggestion
{% Aside %}
To avoid a frustrating experience of auto-reauthentication immediately after a
user has signed out, it's recommended to design a sign-out flow that would prevent that.

FedCM has a 10-minute quiet period after an auto-reauthentication to
prevent this behavior. We are also exploring other approaches to achieve this
such as using
[`CredentialsContainer.preventSilentAccess()`](https://developer.mozilla.org/docs/Web/API/CredentialsContainer/preventSilentAccess).

{% endAside %}
```